src/Engine/Objects/Player.cpp

```diff
@@ -249,6 +248,10 @@ bool Player::CanCastSpell(unsigned int uRequiredMana) {
     return false;
 }
 
+void Player::CastSpellModifyMana(unsigned int uRequiredMana) {
```

Comment:
I'd suggest naming this function `SpendMana`.

Also please add this to the implementation:
`assert(sMana >= uRequiredMana);`

This way we'll catch related bugs if there are any.